src/State.cpp

```diff
@@ -123,7 +124,58 @@ prom_id_t tracer_state_t::enclosing_promise_id() {
          iterator != full_stack.rend(); ++iterator) {
         auto event = *iterator;
         if (event.type == stack_type::PROMISE)
-            return event.promise_id;    
+            return event.promise_id;
     }
     return -1;
 }
+
+std::string to_string(function_type f) {
+    switch (f) {
+        case function_type::CLOSURE:
+            return std::string("CLOSURE");
+        case function_type::BUILTIN:
+            return std::string("BUILTIN");
+        case function_type::SPECIAL:
+            return std::string("SPECIAL");
+        case function_type::TRUE_BUILTIN:
+            return std::string("TRUE BUILTIN");
+        default:
+            // dyntrace_log_warning("unknown function_type %d",
```

Comment:
I guess it should explode if this happens. It's really easy to miss an "ERROR!" value in a large database, unless you're specifically looking for it.

Reply:
I had `dyntrace_error` in all cases.  But it exploded for `stack_type`. I changed it to `dyntrace_warning`, but then I started getting lots of warning messages. Finally, I commented it out temporarily.

Reply:
I will change it to `dyntrace_error`. Ideally we want the tracing to stop as soon as an incorrect value is found.